### ISSUE TITLE
Get hashPrefix from $location

### ISSUE THREAD

**Issue by javarch** [angular/angular.js#15683]
***Note*: for support questions, please use one of these channels: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#question. This repository's issues are reserved for feature requests and bug reports.**

**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behavior?**
Currently, there is no possibility to read the hashPrefix outside of a config-block. So the application needs an own implementation to get the configured hashPrefix in for example a controller or service.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://plnkr.co or similar (template: http://plnkr.co/edit/tpl:yBpEi4).**
-

**What is the expected behavior?**
It would be nice to get a (readonly) property on $location to get the hashPrefix.

**What is the motivation / use case for changing the behavior?**
With the property it is possible to access the hashPrefix without any calculation.

**Which versions of AngularJS, and which browser / OS are affected by this issue? Did this work in previous versions of AngularJS? Please also test with the latest stable and snapshot (https://code.angularjs.org/snapshot/) versions.**
1.6.0 and higher

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**

Thx!

**Comment by Narretz** [comment]
When do you need to programmatically access the hash prefix? I.e. what's the use case?

**Comment by gkalpak** [comment]
I don't think this is a common need and since it is totally (and easily) feasible outside core, it's 👎 from me.

**Comment by javarch** [comment]
The concrete use case is writing a library in which we do not know the hashPrefix set by the developer (config-block of the application is done after the config-block of the depending libraries). 
The library needs to interact with an embedding rich client, which has to set the route-path as full fragment for interaction with the SPA (e.x. #!/page1 or in other applications #/page1).  As the hashPrefix can differ, the library needs to calculate the full fragment from the url. The available routes are then communicated to the richt client so it can change the route via the browser-path of the current application.

Thx!
